src/main/java/io/neonbee/internal/codec/DataExceptionMessageCodec.java

```diff
@@ -0,0 +1,38 @@
+package io.neonbee.internal.codec;
+
+import io.neonbee.data.DataException;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.eventbus.MessageCodec;
+import io.vertx.core.json.JsonObject;
+
+public class DataExceptionMessageCodec implements MessageCodec<DataException, DataException> {
+
+    @Override
+    public void encodeToWire(Buffer buffer, DataException body) {
+        JsonObject json = JsonObject.mapFrom(body);
```

Comment:
Uhhh, really? Do we rely encoding this class with Jackson here?

Are we sure that this is intended? I mean both Vert.x documentation and also Jackson documentation clearly mentions that the mapFrom / encode they do is intended for POJO's only:

https://vertx.io/docs/apidocs/io/vertx/core/json/Json.html#encode-java.lang.Object-

Our DataException is clearly NO POJO! I think we have to invest in an own encoding here, this is prone to failure!